Specify header path in CT_CC_SYSROOT_ARG only for glibc

stephanosio opened the pull request:
When building a non-sysrooted toolchain, `--with-headers` argument is specified in `CT_CC_SYSROOT_ARG` as a hack, supposedly because "final gcc will define disable_glibc while building libgcc, and you'll have no profiling."

This, however, leads to `--with-headers` being specified multiple times when building libstdc++ for additional libc variants (e.g. newlib-nano and picolibc) and results in wrong libc headers being used by the libstdc++ build under certain circumstances -- GCC does not use the last specified `--with-headers` when building for macOS and Windows hosts.

Since the above hack is intended for glibc only, this commit adds a check to ensure that `--with-headers` is added to `CT_CC_SYSROOT_ARG` only when building glibc.

In addition, the final gcc build process was relying on `CT_CC_SYSROOT_ARG` to
provide the `--with-headers` argument pointing to the libc header
directory.

Since `CT_CC_SYSROOT_ARG` no longer provides `--with-headers`, this adds
one directly to the final gcc build invocation.

stephanosio committed:
Specify header path in CT_CC_SYSROOT_ARG only for glibc

When building a non-sysrooted toolchain, `--with-headers` argument is
specified in `CT_CC_SYSROOT_ARG` as a hack, supposedly because "final
gcc will define disable_glibc while building libgcc, and you'll have no
profiling."

This, however, leads to `--with-headers` being specified multiple times
when building libstdc++ for additional libc variants (e.g. newlib-nano
and picolibc) and results in wrong libc headers being used by the
libstdc++ build under certain circumstances -- GCC does not use the last
specified `--with-headers` when building for macOS and Windows hosts.

Since the above hack is intended for glibc only, this commit adds a
check to ensure that `--with-headers` is added to `CT_CC_SYSROOT_ARG`
only when building glibc.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

keith-packard reviewed (approved):
This looks correct to me.

stephanosio committed:
gcc: Specify '--with-headers' when building final gcc

The final gcc build process was relying on `CT_CC_SYSROOT_ARG` to
provide the `--with-headers` argument pointing to the libc header
directory.

Since `CT_CC_SYSROOT_ARG` no longer provides `--with-headers`, this adds
one directly to the final gcc build invocation.

Without this, gcc build system will not copy all the required libc
headers into the `sys-include` directory.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

keith-packard commented on scripts/build/cc/gcc.sh:
```diff
@@ -280,6 +280,7 @@ do_gcc_core_backend() {
         gcc_build|gcc_host)
             CT_DoLog EXTRA "Configuring final gcc compiler"
             extra_config+=( "${CT_CC_SYSROOT_ARG[@]}" )
+            extra_config+=( "--with-headers=${CT_PREFIX_DIR}/${CT_TARGET}/include" )
```
This piece doesn't seem motivated by the series description?

stephanosio replied:
I have updated the series/PR description.

keith-packard replied:
Thanks, this seems a lot more clear now. Happy to approve when you've switched it away from Draft.

keith-packard reviewed (approved):
Very clear descriptions now. Best commit of the day award.
